Validator failed for v0.71.3-rc.1 "MerkleSynchronizationException: Wasn't able to finish full rehashing in time"

Validator (latest hedera-state-validator-0.72.0-SNAPSHOT-all.jar)

failed unexpectedly against v0.71.3-rc.1 with error:


Rehash.reHash,reHash(DeserializedSignedState) - FAILED, time taken - 668 sec
org.junit.jupiter.api.extension.ParameterResolutionException: Failed to resolve parameter [com.swirlds.platform.state.snapshot.DeserializedSignedState arg0] in method [void com.hedera.statevalidation.validator.state.Rehash.reHash(com.swirlds.platform.state.snapshot.DeserializedSignedState) throws java.lang.Exception]: com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException: Wasn't able to finish full rehashing in time
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.RuntimeException: com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException: Wasn't able to finish full rehashing in time
	at com.hedera.statevalidation.util.StateUtils.initState(StateUtils.java:210)
	at com.hedera.statevalidation.util.StateUtils.getDeserializedSignedState(StateUtils.java:176)
	at com.hedera.statevalidation.util.StateUtils.getDeserializedSignedState(StateUtils.java:166)
	at com.hedera.statevalidation.util.junit.DeserializedSignedStateResolver.resolveParameter(DeserializedSignedStateResolver.java:25)
	at com.hedera.statevalidation.util.junit.DeserializedSignedStateResolver.resolveParameter(DeserializedSignedStateResolver.java:14)
	... 2 more
Caused by: com.swirlds.common.merkle.synchronization.utility.MerkleSynchronizationException: Wasn't able to finish full rehashing in time
	at com.swirlds.virtualmap.VirtualMap.fullLeafRehashIfNecessary(VirtualMap.java:589)
	at com.swirlds.virtualmap.VirtualMap.loadFromDirectory(VirtualMap.java:1638)
	at com.swirlds.state.merkle.VirtualMapStateLifecycleManager.loadSnapshot(VirtualMapStateLifecycleManager.java:212)
	at com.swirlds.platform.state.snapshot.SignedStateFileReader.readState(SignedStateFileReader.java:74)
	at com.hedera.statevalidation.util.StateUtils.initState(StateUtils.java:201)
	... 6 more
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at com.swirlds.virtualmap.VirtualMap.fullLeafRehashIfNecessary(VirtualMap.java:579)
	... 10 more
TokenRelationsIntegrity.validate,validate(VirtualMapState) - SUCCEEDED, time taken - 728 sec
AccountValidator.validate,validate(VirtualMapState) - SUCCEEDED, time taken - 88 sec
ValidateInternalIndex.validateIndex,validateIndex(VirtualMapState) - SUCCEEDED, time taken - 304 sec
ValidateLeafIndex.validateIndex,validateIndex(VirtualMapState) - SUCCEEDED, time taken - 522 sec


Latitude/Dallas: solo-mdlt-n4

Full log: https://perf.analytics.eng.hashgraph.io/ephemeral/test_runs/validator.log

--- a/hedera-state-validator/src/main/java/com/hedera/statevalidation/util/ConfigUtils.java
+++ b/hedera-state-validator/src/main/java/com/hedera/statevalidation/util/ConfigUtils.java
@@ -119,7 +119,8 @@ public final class ConfigUtils {
                 .withSource(new SimpleConfigSource().withValue("merkleDb.minNumberOfFilesInCompaction", 2))
                 .withSource(new SimpleConfigSource().withValue("merkleDb.maxFileChannelsPerFileReader", FILE_CHANNELS))
                 .withSource(new SimpleConfigSource().withValue("merkleDb.maxThreadsPerFileChannel", 1))
-                .withSource(new SimpleConfigSource().withValue("virtualMap.fullRehashTimeoutMs", FULL_REHASH_TIMEOUT_MS))
+                .withSource(
+                        new SimpleConfigSource().withValue("virtualMap.fullRehashTimeoutMs", FULL_REHASH_TIMEOUT_MS))
                 .withConverter(CongestionMultipliers.class, new CongestionMultipliersConverter())
                 .withConverter(EntityScaleFactors.class, new EntityScaleFactorsConverter())
                 .withConverter(KnownBlockValues.class, new KnownBlockValuesConverter())

